site/app/views/submission/HomeworkView.php

```diff
@@ -1030,6 +1030,16 @@ private function renderVersionBox(GradedGradeable $graded_gradeable, $version_in
             }
         }
 
+        $failed_file = '';
+        // See if the grade has succeeded or failed
+        foreach ($param['files'] as $file) {
+            if (str_contains($file['relative_name'], 'failed')) {
```

Comment:
Single file containing "failed" will generate amount of error message, but we discussed on the meeting and should not worry too much about it.